BDArmory/Ammo/ModuleCASE.cs

```diff
@@ -440,56 +444,78 @@ void OnDestroy()
 
         void OnGUI()
         {
-            if (HighLogic.LoadedSceneIsEditor && BDArmorySettings.BD_AMMOBINS)
+            if (HighLogic.LoadedSceneIsEditor)
             {
-                if (BDArmorySetup.showWeaponAlignment)
-                    DrawDetonationVisualization();
-                else
+                if (BDArmorySettings.BD_AMMOBINS) //having this on showWeaponAlignment could get really annoying if lots of ammo boxes on a craft and merely wanting to calibrate guns
```

Comment:
I noticed that the visualisation wasn't resetting properly when toggling the simulation/`BD_AMMOBINS`. This patch fixes that. Also, I didn't see any reason for the `globalToggle` parameter since it's just using `BDArmorySetup.showCASESimulation`.
```diff
diff --git a/BDArmory/Ammo/ModuleCASE.cs b/BDArmory/Ammo/ModuleCASE.cs
index 15ac15201..ec62ec67e 100644
--- a/BDArmory/Ammo/ModuleCASE.cs
+++ b/BDArmory/Ammo/ModuleCASE.cs
@@ -448,2 +448,3 @@ void OnGUI()
             {
+                bool disableCASESimulation = false;
                 if (BDArmorySettings.BD_AMMOBINS) //having this on showWeaponAlignment could get really annoying if lots of ammo boxes on a craft and merely wanting to calibrate guns
@@ -451,5 +452,7 @@ void OnGUI()
                     if (BDArmorySetup.showCASESimulation || blastSim >= 1)
-                    DrawDetonationVisualization(BDArmorySetup.showCASESimulation); //though perhaps a per-box visualizer toggle would be smarter than a global one?
+                        DrawDetonationVisualization(); //though perhaps a per-box visualizer toggle would be smarter than a global one?
+                    else if (blastTimeline > 0) disableCASESimulation = true;
                 }
-                else if (simStartTime > 0)
+                else if (blastTimeline > 0) disableCASESimulation = true;
+                if (disableCASESimulation)
                 {
@@ -468,3 +471,3 @@ void OnGUI()
 
-        void DrawDetonationVisualization(bool globalToggle)
+        void DrawDetonationVisualization()
         {
@@ -482,4 +485,5 @@ void DrawDetonationVisualization(bool globalToggle)
             GetBlastRadius();
-            if (simTimer > 5) simStartTime = Time.time; //another possible improvement would have a 'sim blast range' slider that would allow seeing damage at specific range instead of cycling the anim
-            blastTimeline = globalToggle ? Mathf.Clamp01(simTimer / 2) : blastSim / 100;
+            if (!BDArmorySetup.showCASESimulation) simStartTime = 0;
+            else if (simTimer > 5) simStartTime = Time.time; //another possible improvement would have a 'sim blast range' slider that would allow seeing damage at specific range instead of cycling the anim
+            blastTimeline = BDArmorySetup.showCASESimulation ? Mathf.Clamp01(simTimer / 2) : blastSim / 100;
             float blastDmg = Mathf.Clamp(blastCurve.Evaluate(blastRadius * blastTimeline) + (11 - (blastRadius * blastTimeline * 0.4f)) * (float)ammoExplosionYield, 0, 1200) / (CASELevel == 1 ? 2 : 1); //CASE I clamps to 600, so mult CAS 0 dmg to maintian color per x dmg value
```